src/rgw/rgw_op.h

```diff
@@ -146,31 +146,31 @@ class RGWGetObj : public RGWOp {
 
   int init_common();
 public:
-  RGWGetObj() {
-    range_str = NULL;
-    if_mod = NULL;
-    if_unmod = NULL;
-    if_match = NULL;
-    if_nomatch = NULL;
-    mod_zone_id = 0;
-    mod_pg_ver = 0;
-    start = 0;
-    ofs = 0;
-    total_len = 0;
-    end = -1;
-    mod_ptr = NULL;
-    unmod_ptr = NULL;
-    get_data = false;
-    partial_content = false;
-    range_parsed = false;
-    skip_manifest = false;
-    is_slo = false;
-    first_block = 0;
-    last_block = 0;
-    q_ofs = 0;
-    q_len = 0;
-    first_data = true;
-    cur_ofs = 0;
+  RGWGetObj(): 
+    range_str(nullptr),
+    if_mod(nullptr),
+    if_unmod(nullptr),
+    if_match(nullptr),
+    if_nomatch(nullptr),
+    mod_zone_id(0),
+    mod_pg_ver(0),
+    ofs(0),
+    total_len(0),
+    start(0),
+    end(-1),
+    mod_ptr(nullptr),
+    unmod_ptr(nullptr),
+    get_data(false),
+    partial_content(false),
+    range_parsed(false),
+    skip_manifest(false),
+    is_slo(false),
+    first_block(0),
+    last_block(0),
+    q_ofs(0),
+    q_len(0),
+    first_data(true),
+    cur_ofs(0) {
```

Comment:
i would prefer that we use inline initialization where these variables are declared, ie:
```c++
  const char *range_str{nullptr};
...
  uint32_t mod_zone_id{0};
```
this way, we only have to change one line when we add/remove members, and we can add extra constructors without duplicating all of them

(whether you use the `var{foo}` or `var = foo` syntax is up to you)